src/Validator/Hostname.php

```diff
@@ -124,7 +120,7 @@ public function isValid(mixed $value): bool
                         }
 
                         // If every section matched; allow
-                        if($matchesAmount === \count($allowedSections)) {
+                        if ($matchesAmount === \count($allowedSections)) {
                             return true;
                         }
                     }
```

Comment:
I think the last line with return `true` might be risky, maybe we can use this PR to change it. If it return false, we can avoid the last `return false;` statement and minimize the code.

Reply:
The last return true is not regarding foreach, but regarding if. If no allowList is provided, return true. As soon as you go inside the if, it ends with a return false.

Anyway, I rewrote the method a little. The logic is still exactly the same, but instead of having scarry `return true` at the end, it is now under if statement to make the code cleaner. Please let me know if it looks better now.

Reply:
https://github.com/utopia-php/framework/blob/f375033bdbbb96a64ee665fc8ec077611f47840c/src/Validator/Hostname.php#L91-L94